Assets/Scripts/Controller/Pickup Pup/PPGameController.cs

```diff
@@ -181,24 +209,23 @@ public bool TryBuyItem(ShopItem item)
     void buyItem(int value, CurrencyType valueCurrencyType,
         int cost, CurrencyType costCurrencyType)
     {
-        dataController.ChangeCurrencyByType(value, valueCurrencyType);
-        dataController.ChangeCurrencyByType(-cost, costCurrencyType);
+        ConvertCurrency(value, valueCurrencyType, cost, costCurrencyType);
     }
 
     public bool TryAdoptDog(DogDescriptor dog)
     {
-        if(Coins.Amount < dog.CostToAdopt || VacantHomeSlots.Amount <= 0)
+        if(CanAfford(CurrencyType.Coins, dog.CostToAdopt) && CanAfford(CurrencyType.HomeSlots, 1))
         {
-            return false;
-        }
-        AdoptDog(dog);
-        return true;
+            AdoptDog(dog);
+            return true;
+        }        
+        return false;       
     }
 
     void AdoptDog(DogDescriptor dog)
```

Comment:
The problem is dogs have two different types of costs, unless you want me to write TryBuyItem twice? Also, I'm not sure what to write as the "value" of a dog. It would make sense to write it's sell-back value, but that should probably go in the spreadsheet.